### PR TITLE
Fixed url for a fork (4.x branch)

### PR DESCRIPTION
**Pull request by asenyaev**
This PR fixes an issue when the fork repository has another name ([example](https://github.com/opencv/opencv/runs/6064424452?check_suite_focus=true#step:6:11)).

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: .github/workflows/timvx_backend_tests.yml [hunk 1]
```diff
@@ -37,7 +38,7 @@ jobs:
           cd opencv
           git config user.email "opencv.ci"
           git config user.name "opencv.ci"
-          git pull -v "https://github.com/${{ env.PR_AUTHOR }}/opencv" "${{ env.SOURCE_BRANCH_NAME }}" --allow-unrelated-histories
+          git pull -v "https://github.com/${{ env.PR_AUTHOR_FORK }}" "${{ env.SOURCE_BRANCH_NAME }}"
```
Comment (fengyuentau) [inline .github/workflows/timvx_backend_tests.yml:41]: Is this change made for those who have opencv fork but with repo name changed?

Comment (asenyaev) [reply .github/workflows/timvx_backend_tests.yml:41]: Yes, we've already met this issue recently.

Comment (fengyuentau) [reply .github/workflows/timvx_backend_tests.yml:41]: Okay, I will forward this change to my PR of workflow for ARM64 as well.

**Review by fengyuentau — APPROVED** [review]

**Review by asmorkalov — APPROVED** [review]
Thanks!

**Pull request merged**